cdc/cdc.cc

```diff
@@ -0,0 +1,221 @@
+/*
+ * Copyright (C) 2019 ScyllaDB
+ */
+
+/*
+ * This file is part of Scylla.
+ *
+ * Scylla is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Scylla is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Scylla.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <utility>
+
+#include "cdc/cdc.hh"
+#include "database.hh"
+#include "db/config.hh"
+#include "dht/murmur3_partitioner.hh"
+#include "schema.hh"
+#include "schema_builder.hh"
+#include "service/migration_manager.hh"
+#include "service/storage_proxy.hh"
+#include "service/storage_service.hh"
+
+using locator::snitch_ptr;
+using locator::token_metadata;
+using locator::topology;
+using seastar::sstring;
+using service::migration_manager;
+using service::storage_proxy;
+
+namespace cdc {
+
+sstring log_name(const sstring& table_name) {
+    static constexpr auto cdc_log_suffix = "_scylla_cdc_log";
+    return table_name + cdc_log_suffix;
```

Comment:
Those tables will be queried by uses using CQL so characters that are not allowed probably won't work.

Reply:
Bikeshedding, but maybe we should make the suffix `_change_log` to make it clearer.

Reply:
@haaawk I see, I thought the CDC table is an implementation detail and the user will have new instructions to read from it - I completely missed the fact that users will need to read from it directly and understand its inner format to do so. Indeed, if users need to read from it, they'll need to know the table's name. But in that case maybe you should add checking to the CDC creation code that the CDC tables do not previously exist - because they may.

Reply:
@nyh makes sense. I was planning to solve this by making sure that it's impossible to create a table with _scylla_cdc_log and _scylla_cdc_desc suffix manually. I also planned to do it as a separate task that would be done by one of the new employees. Would that be acceptable?

Reply:
@haaawk we can certainly do this later, but please remember this issue somewhere (FIXME, bug tracker, or wherever you save your list of tasks...).
@avikivity are you ok with that long name, "_scylla_cdc_log" (that users will need to be aware of) or do you request that they change it? We can change it later, of course, while this is still an experimental feature.

Reply:
@nyh there's already an issue https://github.com/scylladb/scylla/issues/4991